pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -434,8 +435,13 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[string]b
 
 	for locality, eps := range lbEndpoints {
 		var weight uint32
+		var overflowStatus bool
 		for _, ep := range eps {
-			weight += ep.LoadBalancingWeight.GetValue()
+			weight, overflowStatus = addUint32(weight, ep.LoadBalancingWeight.GetValue())
+		}
+		if overflowStatus {
+			log.Warnf("Sum of localityLbEndpoints weight is overflow: service:%s, port: %d, locality:%s",
```

Comment:
nit: the error message could indicate the size of the int being overflowed (uint32)

Reply:
+1